### PR TITLE
[SMALLFIX] [TACHYON-1700] Add unit tests in tachyon.securty.{UserTest, LoginUserTest}

### PR DESCRIPTION
**Pull request by ifcharming**
https://tachyon.atlassian.net/browse/TACHYON-1700

### CODE REVIEW

File: core/common/src/test/java/tachyon/security/UserTest.java [hunk 1]
```diff
@@ -43,4 +43,31 @@ public void usedInSecurityContextTest() {
...
+  @Test
+  public void realmAsUserName() {
+    // Add new users into Subject.
+    Subject subject = new Subject();
+    subject.getPrincipals().add(new User("admin/admin@EXAMPLE.com"));
+    subject.getPrincipals().add(new User("admin/mbox.example.com@EXAMPLE.com"));
+    subject.getPrincipals().add(new User("imap/mbox.example.com@EXAMPLE.COM"));
+
+    // Fetch added users.
+    User[] users = subject.getPrincipals(User.class).toArray(new User[0]);
```
Comment (aaudiber) [inline core/common/src/test/java/tachyon/security/UserTest.java]: why do we convert to an array here?

File: core/common/src/test/java/tachyon/security/LoginUserTest.java [hunk 2]
```diff
@@ -84,6 +84,25 @@ public void getSimpleLoginUserProvidedByAppTest() throws Exception {
   }
 
   /**
+   * Test whether we can get login user with conf in SIMPLE mode, when a user list is provided by
+   * by the application through configuration.
+   *
+   * @throws Exception thrown if the current user cannot be retrieved
```
Comment (aaudiber) [inline core/common/src/test/java/tachyon/security/LoginUserTest.java]: we usually skip the `@throws` for tests

File: core/common/src/test/java/tachyon/security/UserTest.java [hunk 3]
```diff
@@ -43,4 +43,31 @@ public void usedInSecurityContextTest() {
     Assert.assertEquals(2, users.length);
     Assert.assertNotEquals(users[0], users[1]);
   }
+
+  /**
+   * This test verifies that full realm names format is supported.
```
Comment (aaudiber) [inline core/common/src/test/java/tachyon/security/UserTest.java]: supported in what way?

**Comment by aaudiber** [comment]
Thanks for improving test coverage @ifcharming! I've left a few comments

File: core/common/src/test/java/tachyon/security/UserTest.java (hunk 1 above)
Comment (ifcharming) [inline core/common/src/test/java/tachyon/security/UserTest.java]: No need, updated all.

File: core/common/src/test/java/tachyon/security/LoginUserTest.java (hunk 2 above)
Comment (ifcharming) [inline core/common/src/test/java/tachyon/security/LoginUserTest.java]: Done all.

File: core/common/src/test/java/tachyon/security/UserTest.java (hunk 3 above)
Comment (ifcharming) [inline core/common/src/test/java/tachyon/security/UserTest.java]: Updated comment, thanks!

File: core/common/src/test/java/tachyon/security/UserTest.java [hunk 4]
```diff
@@ -37,10 +39,32 @@ public void usedInSecurityContextTest() {
...
+    Set<User> users = subject.getPrincipals(User.class);
 
     // verification
-    Assert.assertEquals(2, users.length);
-    Assert.assertNotEquals(users[0], users[1]);
+    Assert.assertEquals(2, users.size());
+  }
+
+  /**
+   * This test verifies that full realm format is valid as User name.
```
Comment (aaudiber) [inline core/common/src/test/java/tachyon/security/UserTest.java]: `{@link User}`

File: core/common/src/test/java/tachyon/security/UserTest.java [hunk 5]
```diff
@@ -37,10 +39,32 @@ public void usedInSecurityContextTest() {
...
+   */
+  @Test
+  public void realmAsUserName() {
```
Comment (aaudiber) [inline core/common/src/test/java/tachyon/security/UserTest.java]: We generally end tests with "Test", e.g. fooTest, barTest

**Comment by aaudiber** [comment]
I left a couple more minor comments

**Comment by aaudiber** [comment]
sorry, close was accidental

File: core/common/src/test/java/tachyon/security/UserTest.java (hunk 4 above)
Comment (ifcharming) [inline core/common/src/test/java/tachyon/security/UserTest.java]: Done.

File: core/common/src/test/java/tachyon/security/UserTest.java (hunk 5 above)
Comment (ifcharming) [inline core/common/src/test/java/tachyon/security/UserTest.java]: Done.

**Comment by aaudiber** [comment]
LGTM

**Comment by apc999** [comment]
+1 LGTM

**Pull request merged**